subprojects/logging/src/main/java/org/gradle/internal/logging/sink/GroupingProgressLogEventGenerator.java

```diff
@@ -84,6 +103,19 @@ private void onStart(ProgressStartEvent startEvent) {
             if (isGroupedOperation(startEvent.getBuildOperationCategory())) {
                 String header = startEvent.getLoggingHeader() != null ? startEvent.getLoggingHeader() : startEvent.getDescription();
                 operationsInProgress.put(buildOpId, new OperationGroup(startEvent.getCategory(), header, startEvent.getTimestamp(), startEvent.getBuildOperationId()));
+
+                if (future == null || future.isCancelled()) {
+                    future = executor.scheduleWithFixedDelay(new Runnable() {
```

Comment:
The thing is `OperationGroup` can be started at a various time. For example, one at 0s, another one at 2s and another one at 4s. If we use fixed rate, the output won't follow a 5s delay before being print to the console (first one will be 5s delay, the second one will be 3s delay and the last one will be 1s delay.

We start the scheduling when the first `OperationGroup` is started so we know we can delay by 5s. Afterward, we keep track of the delay by periodically (500ms) checking the last update time compared to the current time. This ensures we keep a buffer of 5s for every task.